addons/survey/models/survey_user_input.py

```diff
@@ -878,7 +878,7 @@ def _get_answer_score_values(self, vals, compute_speed_score=True):
         if compute_speed_score and answer_score > 0:
             user_input = self.env['survey.user_input'].browse(user_input_id)
             session_speed_rating = user_input.exists() and user_input.is_session_answer and user_input.survey_id.session_speed_rating
-            if session_speed_rating:
+            if session_speed_rating and question.is_time_limited:
```

Comment:
Shouldn't we also check that question.time_limit is set to be sure (you never know how people end up with fucked up configuration) ?